File: system-test/typescript/src/testkit/dsl/port/given/steps/given-step.ts
import type { GivenStage } from '../given-stage.js';
import type { WhenStage } from '../../when/when-stage.js';
import type { ThenStage } from '../../then/then-stage.js';

export interface GivenStep {
  and(): GivenStage;
  when(): WhenStage;
  then(): ThenStage;
}

File: system/multitier/backend-java/src/testSupport/java/com/mycompany/myshop/backend/testkit/dsl/port/given/steps/base/GivenStep.java
package com.mycompany.myshop.backend.testkit.dsl.port.given.steps.base;

import com.mycompany.myshop.backend.testkit.dsl.port.given.GivenStage;
import com.mycompany.myshop.backend.testkit.dsl.port.then.ThenStage;
import com.mycompany.myshop.backend.testkit.dsl.port.when.WhenStage;

public interface GivenStep {
    GivenStage and();

    WhenStage when();

    ThenStage then();
}
